src/game/gui/trnselect.c

```diff
@@ -0,0 +1,178 @@
+#include "game/gui/trnselect.h"
+#include "formats/error.h"
+#include "formats/palette.h"
+#include "formats/tournament.h"
+#include "game/gui/label.h"
+#include "game/gui/widget.h"
+#include "resources/pathmanager.h"
+#include "resources/sprite.h"
+#include "resources/trnmanager.h"
+#include "utils/allocator.h"
+#include "utils/list.h"
+#include "utils/log.h"
+#include "video/video.h"
+
+// Local small gauge type
+typedef struct {
+    sprite *img;
+    list *tournaments;
+    component *label;
+    int max;
+    int selected;
+    palette palette_backup;
+} trnselect;
+
+static void trnselect_render(component *c) {
+    trnselect *g = widget_get_obj(c);
+
+    video_render_sprite(g->img->data, c->x + g->img->pos.x, c->y + g->img->pos.y, BLEND_ALPHA, 0);
+    if(g->label) {
+        component_render(g->label);
+    }
+}
+
+void load_description(component **c, const char *desc) {
+    int width = 320, center = 0, vmove = 0, size = -1, color = -1, x = 0;
+    const char *end = desc;
+    char *ptr = NULL;
+    // don't judge me!
```

Comment:
We might want to split this parser to it's own thing later on. I think there were some other places that used this also?

Reply:
Yes, the help/about stuff also uses it